packages/explat-client/src/internal/experiment-assignment-store.ts

```diff
@@ -4,46 +4,81 @@
 import type { ExperimentAssignment } from '../types';
 import * as Validations from './validations';
 
+const localStorage =
+	typeof window !== 'undefined' && window.localStorage
+		? window.localStorage
+		: // LocalStorage polyfill from https://gist.github.com/juliocesar/926500
+		  {
+				_data: {} as Record< string, string >,
+				setItem: function ( id: string, val: string ) {
+					return ( this._data[ id ] = String( val ) );
+				},
```

Comment:
* We don't need the string conversion since we're using TypeScript and we control the uses of this function.
* The return value for `setItem` is `undefined`: https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem#return_value

```suggestion
				setItem: function ( id: string, val: string ) {
					this._data[ id ] = val;
				},
```